NoSuchMethodError error raised by HttpStateChannel  with Jetty 12.1.6 run with maven plugin

**Jetty version(s)**

Jetty 12.1.6

**Jetty Environment**

ee10, ee11

**HTTP version**
<!-- HTTP 1.0, 1.1, 2 or 3 -->

**Java version/vendor** `(use: java -version)`

openjdk version "21.0.6" 2025-01-21 LTS
OpenJDK Runtime Environment Temurin-21.0.6+7 (build 21.0.6+7-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.6+7 (build 21.0.6+7-LTS, mixed mode, sharing)

**OS type/version**

Ubuntu 24.04.3 LTS

**Description**
<!-- Do not report security issues here! See [Jetty Security Reports](https://jetty.org/security.html) -->

In Vaadin Flow, we use Jetty Maven plugin in EMBED mode to run integration tests.
We noticed that bumping from Jetty 12.1.5 to 12.1.6 caused lots of failures in our test suite.
One of the issues seems to be caused by the usage of SLF4J `atDebug` method in `HttpChannelState`, introduced in 12.1.6.
Even if `atDebug` is a valid SLF4J 2 API, and our product uses the latest 2.x, we noticed that at runtime `HttpStateChannel.LOG` is an instance of `org.slf4j.impl.MavenSimpleLogger` that seems to be a SLF4J 1.x implementation, missing the `atDebug` method.

Happens both with Maven 3.8.7 and 3.9.11

```
java.lang.NoSuchMethodError: 'org.slf4j.spi.LoggingEventBuilder org.slf4j.Logger.atDebug()'
    at org.eclipse.jetty.server.internal.HttpChannelState.completeStream (HttpChannelState.java:723)
    at org.eclipse.jetty.server.internal.HttpChannelState$ChannelCallback.completeChannelCallback (HttpChannelState.java:1775)
    at org.eclipse.jetty.server.internal.HttpChannelState$ChannelCallback.succeeded (HttpChannelState.java:1603)
    at org.eclipse.jetty.ee10.servlet.ServletChannel.onCompleted (ServletChannel.java:809)
    at org.eclipse.jetty.ee10.servlet.ServletChannel.handle (ServletChannel.java:439)
    at org.eclipse.jetty.server.handler.ContextHandler$ScopedContext.run (ContextHandler.java:1697)
    at org.eclipse.jetty.server.handler.ContextHandler$ScopedContext.run (ContextHandler.java:1684)
    at org.eclipse.jetty.ee10.servlet.ServletChannelState.runInContext (ServletChannelState.java:1311)
    at org.eclipse.jetty.ee10.servlet.ServletChannelState.complete (ServletChannelState.java:852)
    at org.eclipse.jetty.ee10.servlet.AsyncContextState.complete (AsyncContextState.java:61)
    at org.atmosphere.container.Servlet30CometSupport.endAsyncContext (Servlet30CometSupport.java:129)
    at org.atmosphere.container.Servlet30CometSupport.action (Servlet30CometSupport.java:109)
    at org.atmosphere.container.Servlet30CometSupport.action (Servlet30CometSupport.java:45)
    at org.atmosphere.cpr.AtmosphereResourceImpl.resume (AtmosphereResourceImpl.java:302)
    at org.atmosphere.handler.AbstractReflectorAtmosphereHandler.postStateChange (AbstractReflectorAtmosphereHandler.java:219)
    at org.atmosphere.handler.AbstractReflectorAtmosphereHandler.onStateChange (AbstractReflectorAtmosphereHandler.java:162)
    at com.vaadin.flow.server.communication.PushAtmosphereHandler.onStateChange (PushAtmosphereHandler.java:54)
    at org.atmosphere.cpr.DefaultBroadcaster.invokeOnStateChange (DefaultBroadcaster.java:1036)
    at org.atmosphere.cpr.DefaultBroadcaster.prepareInvokeOnStateChange (DefaultBroadcaster.java:1056)
    at org.atmosphere.cpr.DefaultBroadcaster.executeAsyncWrite (DefaultBroadcaster.java:870)
    at org.atmosphere.cpr.DefaultBroadcaster$2.run (DefaultBroadcaster.java:477)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:572)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
```


**How to reproduce?**

Unfortunately, I could not make a simple reproducer, but here's the link to the [PR](https://github.com/vaadin/flow/pull/23382) that bumps the Jetty version.
I'll try to create a reproducible example later.

I don't know if this can be a good way for you to verify the issue, but I was able to reproduce it on the jetty repository.
I cloned the repo, built it (fast build) and ran `mvn verify` in the `jetty-ee10/jetty-ee10-maven-plugin` module; all the tests passed.

Then I modified `jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java` adding the following instruction in the main constructor

```java
    public Server(@Name("threadPool") ThreadPool threadPool, @Name("scheduler") Scheduler scheduler, @Name("bufferPool") ByteBufferPool bufferPool)
    {
        LOG.atInfo().log("=========================== Creating Server ===========================");
        ...
    }
```

After that, I rebuilt everything (fast build again) and re-run the tests in `jetty-ee10/jetty-ee10-maven-plugin`  with `mvn clean verify -Dmaven.build.cache.enabled=false`.
This time all tests failed, and in the logs I could see errors like

```
[ERROR] Failed to execute goal org.eclipse.jetty.ee10:jetty-ee10-maven-plugin:12.1.7-SNAPSHOT:start (start-jetty) on project jetty-simple-webapp: Execution start-jetty of goal org.eclipse.jetty.ee10:jetty-ee10-maven-plugin:12.1.7-    SNAPSHOT:start failed: An API incompatibility was encountered while executing org.eclipse.jetty.ee10:jetty-ee10-maven-plugin:12.1.7-SNAPSHOT:start: java.lang.NoSuchMethodError: 'org.slf4j.spi.LoggingEventBuilder org.slf4j.Logger.atInfo()'
```

--- a/pom.xml
+++ b/pom.xml
@@ -327,7 +327,7 @@
     <osgi-util-version>3.7.400</osgi-util-version>
     <osgi-util-xml-version>1.0.2</osgi-util-xml-version>
     <osgi-version>3.24.0</osgi-version>
-    <osgi.slf4j.import.packages>org.slf4j;version="[1.7,3.0)", org.slf4j.event;version="[1.7,3.0)", org.slf4j.helpers;version="[1.7,3.0)", org.slf4j.spi;version="[1.7,3.0)"</osgi.slf4j.import.packages>
+    <osgi.slf4j.import.packages>org.slf4j;version="[2.0,3.0)", org.slf4j.event;version="[2.0,3.0)", org.slf4j.helpers;version="[2.0,3.0)", org.slf4j.spi;version="[2.0,3.0)"</osgi.slf4j.import.packages>
     <pax.exam.version>4.14.0</pax.exam.version>
     <pax.url.version>3.0.2</pax.url.version>
     <plexus-classworlds.version>2.9.0</plexus-classworlds.version>

